### PR TITLE
pscan Beta to Release Promotions

### PR DESCRIPTION
**Pull request by kingthorin**
See:
zaproxy/zaproxy#4453
zaproxy/zaproxy#4455
zaproxy/zaproxy#4456
zaproxy/zaproxy#4457
zaproxy/zaproxy#4458
zaproxy/zaproxy#4459

### CODE REVIEW

**Comment by kingthorin** [comment]
I left this as individual commits in order to simplify review.

I'm glad to flatten and add fix tags once it's been looked at.

File: addOns/pscanrules/src/main/resources/org/zaproxy/zap/extension/pscanrules/resources/Messages.properties [hunk 1]
```diff
@@ -16,12 +23,39 @@ pscanrules.charsetmismatch.extrainfo.html.header_metacharset_mismatch=There was
 pscanrules.charsetmismatch.extrainfo.html.metacontenttype_metacharset_mismatch=There was a charset mismatch between the META charset and the META content-type encoding declaration: [{0}] and [{1}] do not match.
 pscanrules.charsetmismatch.extrainfo.xml=There was a charset mismatch between the HTTP Header and the XML encoding declaration: [{0}] and [{1}] do not match.
 
-pscanrules.headerxssprotectionscanner.name = Web Browser XSS Protection Not Enabled
```
Comment (thc202) [inline addOns/pscanrules/src/main/resources/org/zaproxy/zap/extension/pscanrules/resources/Messages.properties]: This change could be reverted.

File: addOns/pscanrulesBeta/CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,12 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - Fix typo in scanner name.
 - Tweak alert Authentication Credentials Captured to use Description field instead of Attack.
 - Remove Charset Mismatch Scanner (promoted to release Issue 4460).
+- Remove ViewState Scanner (promoted to release Issue 4453).
+- Remove Insecure JSF ViewState Scanner (promoted to release Issue 4455).
+- Remove Insecure Authentication Scanner (promote to release Issue 4456).
```
Comment (thc202) [inline addOns/pscanrulesBeta/CHANGELOG.md]: `promoted` like the above? (Same comment for the following.)

**Comment by thc202** [comment]
I'm fine with the individual commits (although last one could be "fixed up", we should add a test to assert that the file exists).

File: addOns/pscanrules/src/main/resources/org/zaproxy/zap/extension/pscanrules/resources/Messages.properties (hunk 1 above)
Comment (kingthorin) [reply addOns/pscanrules/src/main/resources/org/zaproxy/zap/extension/pscanrules/resources/Messages.properties]: I just noticed the white space change, oops.

**Comment by kingthorin** [comment]
Ok I addressed the feedback. Will add a test for the file shortly.

**Comment by kingthorin** [comment]
Hmm ok so how do I test for resource files, relative to the unit test? Because for the purposes of the unit test the file doesn't exist (which must be why the existing unit tests use a bogus temp file.)

**Comment by thc202** [comment]
I was thinking just test that the file exists in the current directory not when deployed to the ZAP home, e.g.:
```
// Given
String debugErrorFilePath = "/xml/debug-error-messages.txt";
// When
URL debugErrorFile = getClass().getResource(debugErrorFilePath);
// Then
assertThat(debugErrorFile, notNullValue());
```

**Comment by kingthorin** [comment]
I’ll try that out, thanks.

**Comment by kingthorin** [comment]
That doesn't work, that path doesn't exist for the UnitTest.

**Comment by thc202** [comment]
It works fine here (command line and Eclipse), could you provide more details? Is the path starting with a forward slash?

No, just adding that code excerpt to a test method should work.

**Comment by kingthorin** [comment]
image

**Comment by thc202** [comment]
Interesting, is the `zapHomeFiles` directory shown as a package? (And the file is under `zap-extensions/addOns/pscanrules/bin/main/xml/`? It should be automatically copied there.)

Does the test also fail when running from the command line?

**Comment by kingthorin** [comment]
Should I be using Run as “gradle test” or “junit test”? Maybe that’s the issue?

**Comment by kingthorin** [comment]
Oh, nope zapHomeFiles isn’t showing as a package. It’s just in src/main/zapHomeFiles

**Comment by thc202** [comment]
Might need a refresh?

**Comment by kingthorin** [comment]
Bingo! After a Gradle Refresh it junit tested successfully (and now shows as a package).

**Comment by thc202** [comment]
LGTM.

File: addOns/pscanrulesBeta/src/main/javahelp/org/zaproxy/zap/extension/pscanrulesBeta/resources/help/contents/pscanbeta.html [hunk 3]
```diff
@@ -10,25 +10,6 @@
 <H1>Passive Scan Rules - Beta</H1>
 The following beta quality passive scan rules are included in this add-on:
 
-<H2>Cookie - Loosely Scoped</H2>
```
Comment (psiinon) [inline addOns/pscanrulesBeta/src/main/javahelp/org/zaproxy/zap/extension/pscanrulesBeta/resources/help/contents/pscanbeta.html]: Do we want to keep the headings but just say that they've been promoted to Release or will that just make it look messy?

Comment (kingthorin) [reply addOns/pscanrulesBeta/src/main/javahelp/org/zaproxy/zap/extension/pscanrulesBeta/resources/help/contents/pscanbeta.html]: I think that will just end up being clutter'ish down the line, especially with some items being promoted twice. It'll be in the release notes (change log) and git history.

@thc202 what do you think?

Comment (thc202) [reply addOns/pscanrulesBeta/src/main/javahelp/org/zaproxy/zap/extension/pscanrulesBeta/resources/help/contents/pscanbeta.html]: I think it's fine as is

**Comment by thc202** [comment]
Thanks!

(The title tweaks were to unstuck the WIP check.)

**Pull request merged**